### PR TITLE
Added authentik provider

### PR DESCRIPTION
**Pull request by Pacerino**
## Reasoning 💡
I've added [Authentik](https://goauthentik.io/) as a provider

## Checklist 🧢
The provider has been added as a `.ts` file, ~~The Documentation about this provider will be added as a PR to the docs repo.~~
_Edit:_ Documentation for the authentik provider has been supplied: https://github.com/nextauthjs/docs/pull/199

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

The provider can be merged as soon its reviewed. I've deployed a local setup and checked if the Login/Logout works as intended.

### CODE REVIEW

File: src/providers/authentik.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  P extends Record<string, any> = AuthentikProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "authentik",
+    name: "Authentik",
+    wellKnown: `${options.issuer}/.well-known/openid-configuration`,
+    type: "oauth",
+    authorization: { params: { scope: "openid email profile" } },
+    checks: ["pkce", "state"],
+    idToken: true,
```
Comment (balazsorban44) [inline src/providers/authentik.ts:34]:
```suggestion
```

This should not be necessary anymore. We detect if `openid` is added as a scope instead.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thank you, this looks great! Left only a small suggestion.


Also, could you paste in an image of sorts that shows you that the login was successful with the name email profile picture? We use it as proof for future reference. :pray:

**Comment by Pacerino** [comment]
I've removed the `idToken`.
Here is a gif as a proof of the working login.
pv0a6

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thank you!

**Pull request merged**